pinot-core/src/main/java/org/apache/pinot/core/util/SegmentProcessorAvroUtils.java

```diff
@@ -83,45 +88,90 @@ public static Schema convertPinotSchemaToAvroSchema(org.apache.pinot.spi.data.Sc
     for (FieldSpec fieldSpec : orderedFieldSpecs) {
       String name = fieldSpec.getName();
       DataType storedType = fieldSpec.getDataType().getStoredType();
+
+      SchemaBuilder.BaseFieldTypeBuilder<Schema> type;
+      if (fieldSpec.isNullable()) {
+        type = fieldAssembler.name(name).type().nullable();
+      } else {
+        type = fieldAssembler.name(name).type();
+      }
+
+      String logicalType = "pinot." + fieldSpec.getDataType().toString().toLowerCase(Locale.US);
       if (fieldSpec.isSingleValueField()) {
         switch (storedType) {
           case INT:
-            fieldAssembler = fieldAssembler.name(name).type().intType().noDefault();
+            fieldAssembler = type.intBuilder()
```

Comment:
Do we need to modify types other than `BIG_DECIMAL`?

Reply:
Probably not. We can discuss whether adding the local type is a good idea.